components/brave_new_tab_ui/components/default/editTopSite/index.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import {
+  StyledDialogWrapper,
+  StyledDialog,
+  DialogTitle,
+  StyledCloseIcon,
+  StyledInputLabel,
+  StyledInput,
+  StyledButtonsContainer,
+  StyledButton
+} from './style'
+import { getLocale } from '../../../../common/locale'
+import CloseIcon from './assets/close-icon'
+
+interface Props {
+  showEditTopSite: boolean
+  targetTopSiteForEditing?: NewTab.Site
+  textDirection: string
+  onClose: () => void
+  onSave: (title: string, url: string, newUrl: string) => void
+}
+
+export default class EditTopSite extends React.PureComponent<Props, {}> {
+  editTopSiteDialogRef: React.RefObject<any>
+  nameInputRef: React.RefObject<any>
+  urlInputRef: React.RefObject<any>
+  constructor (props: Props) {
+    super(props)
+    this.editTopSiteDialogRef = React.createRef()
+    this.nameInputRef = React.createRef()
+    this.urlInputRef = React.createRef()
+  }
+
+  componentDidMount () {
+    document.addEventListener('mousedown', this.handleClickOutside)
+    document.addEventListener('keydown', this.onKeyPressSettings)
+  }
+
+  componentWillUnmount () {
+    document.removeEventListener('mousedown', this.handleClickOutside)
+    document.removeEventListener('keydown', this.onKeyPressSettings)
+  }
+
+  onKeyPressSettings = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      this.props.onClose()
+    } else if (event.key === 'Enter') {
+      if (this.urlInputRef.current && this.urlInputRef.current.value) {
+        this.saveNewTopSite()
+      }
+    }
+  }
```

Comment:
We don't need some of this code if we use [the `<dialog>` html element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog). With that, we'll also gain accessibility improvements. You would call `.showModal()` on the dialog element at `componentDidMount`.
You'll still need to keep the click handler to close the dialog when user clicks the backdrop, if we still want that. In that case you would check if the click is not inside the dialog content element. But `escape` press is handled automatically.

Reply:
How about this as a separate issue? currently no idea for changing current impl to `dialog` :)